scripts/verify-contracts.py

```diff
@@ -0,0 +1,64 @@
+from pathlib import Path
+import ape
+import click
+import structlog
+from eth_utils import to_checksum_address
+import beamer.deploy.artifacts
+import beamer.deploy.config
+from beamer.typing import ChainId
+
+
+log = structlog.get_logger(__name__)
+
+
+def _get_contract_info(artifacts_dir: Path) -> dict[ChainId, dict[str, dict[str, str]]]:
+    contract_info: dict[ChainId, dict[str, dict[str, str]]] = {}
+    for file_path in artifacts_dir.glob("*.deployment.json"):
+        deployment = beamer.deploy.artifacts.Deployment.from_file(file_path)
+        assert deployment.chain is not None
+        chain_id = deployment.chain.chain_id
+        if chain_id not in contract_info:
+            contract_info[chain_id] = {}
+        for contract_name, contract in deployment.chain.contracts.items():
+            if file_path.stem == "base.deployment":
+                ecosystem = "ethereum"
+            else:
+                _, ecosystem = file_path.stem.split(".")[0].split("-")
+            contract_info[chain_id][contract_name] = {
+                "address": contract.address,
+                "ecosystem": ecosystem,
+                "network": file_path.parent.stem,
+            }
+
+    return contract_info
+
+
+@click.command()
+@click.argument("artifacts_dir", type=click.Path(file_okay=False, dir_okay=True, path_type=Path))
+@click.argument("rpc_file", type=click.Path(file_okay=True, dir_okay=False, path_type=Path))
+def main(artifacts_dir: Path, rpc_file: Path) -> None:
+    contract_info = _get_contract_info(artifacts_dir)
+    rpc_info = beamer.deploy.config.load_rpc_info(rpc_file)
+    for chain_id, contracts in contract_info.items():
+        rpc_url = rpc_info[chain_id]
+        for name, contract in contracts.items():
+            ape_ecosystem = getattr(ape.networks, contract["ecosystem"])
+            ape_network = getattr(ape_ecosystem, contract["network"])
+            try:
+                with ape_network.use_provider(rpc_url):
+                    etherscan = ape.networks.provider.network.explorer
+                    assert etherscan is not None
```

Comment:
Nitpick: why do we call it `etherscan` instead of e.g. `explorer`?

Reply:
because it really is etherscan dependent any other explorers like with blockscout or zksync explorer or for startnet this would return None already